### PR TITLE
CNN_model

### PR DESCRIPTION
**Pull request by yoplnaa**
## 🎯 PR 유형

- [x] ✨ 새로운 기능 추가
- [ ] 🐛 버그 수정
- [x] 📝 문서 업데이트
- [x] ♻️ 코드 리팩토링
- [ ] 🎨 코드 스타일 변경
- [ ] 🔧 설정 파일 수정
- [ ] 🔨 CI/CD 관련 변경
- [ ] 🌱 초기 커밋
- [ ] ⚡️ 성능 개선
- [ ] 🔒 보안 이슈 수정

## 🔍 관련 이슈
- close #이슈번호

## 💡 변경 사항

- CNN model 업로드
- sample_data 삭제
- markdown 가독성 향상
- 사용 용어 통일

## ✅ 체크리스트
- [x] 관련 문서를 업데이트하였나요?
- [x] 변경사항에 대한 자체 리뷰를 하였나요?

## 🔄 테스트 방법

1.  주석과 마크 다운을 참고하여 파일 경로에 유의
2.  기타 문의 사항이 있을 시 a01089710746@gmail.com 으로 연락주세요

### CODE REVIEW

**Review by GideokKim — COMMENTED** [review]
sample data는 삭제하시는게 좋을 것 같아요. 대신 sample data를 구축하는 방법이나 구하는 방법을 안내해주면 좋을 것 같습니다!

예를 들어 AI Hub나 kaggle 사이트를 추천해주는 문서 정도면 cnn model을 학습하고자 하시는 분들이 sample data를 잘 찾을 것 같습니다!

File: cnn_test_annotated.ipynb [hunk 1]
```diff
@@ -0,0 +1,853 @@
...
+     "text": [
+      "✅ 데이터셋이 존재하는 폴더: /mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-6a025-ml/code/Users/6a025/cnn_test/data_set\n",
+      "✅ 데이터셋이 존재하는 폴더: /mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-6a025-ml/code/Users/6a025/cnn_test\n",
+      "✅ 데이터셋이 존재하는 폴더: /mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-6a025-ml\n",
+      "❌ 데이터셋 없음: /home/user/data_set\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
```
Comment (GideokKim) [inline cnn_test_annotated.ipynb:157]: 가장 처음 셀에서 `import os`를 했으나 여기서도 import를 하고 있어요. 저는 해도 되고 안해도 된다고 생각하는데 각각 아래와 같은 이유가 있습니다!

1. `import os`를 또 할 필요 없다 -> 이미 위에서 했기 때문에 중복이다.
2. `import os`를 또 하는 것이 좋다 -> 이 코드 셀을 북사 붙여넣기로 사용하는 작업자가 있을 수 있으니 그대로 두자.

다른 분들의 생각은 어떠신가요? @BrianK64 @YiHeeJu @Yoonsoyoung02 @yoplnaa

File: cnn_test_annotated.ipynb [hunk 2]
```diff
@@ -0,0 +1,853 @@
...
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "✅ 총 19개의 음식 클래스가 있습니다.\n"
+     ]
+    }
+   ],
+   "source": [
```
[the remark this answers is not in the document]
Comment (GideokKim) [reply cnn_test_annotated.ipynb:339]: 저는 해도 되고 안해도 된다고 생각하는데 각각 아래와 같은 이유가 있습니다!

import를 또 할 필요 없다 -> 이미 위에서 했기 때문에 중복이다.
import를 또 하는 것이 좋다 -> 이 코드 셀을 북사 붙여넣기로 사용하는 작업자가 있을 수 있으니 그대로 두자.

다른 분들의 생각은 어떠신가요? @BrianK64 @YiHeeJu @Yoonsoyoung02 @yoplnaa

File: cnn_test_annotated.ipynb [hunk 3]
```diff
@@ -0,0 +1,853 @@
...
+    "# 현재 작업 디렉토리 확인 및 데이터 폴더 존재 여부 체크\n",
+    "import os\n",
+    "\n",
+    "print(\"현재 작업 디렉토리:\", os.getcwd())  # 현재 작업 중인 폴더 확인\n",
+    "print(\"Train 폴더 존재 여부:\", os.path.exists(\"./data_set/sample_data/train\")) #해당폴더에 train 데이터 셋이 존재 하면True\n",
+    "print(\"Test 폴더 존재 여부:\", os.path.exists(\"./data_set/sample_data/test\")) #해당폴더에 test 데이터 셋이 존재 하면True\n"
```
[the remark this answers is not in the document]
Comment (yoplnaa) [reply cnn_test_annotated.ipynb:133]: 생각 못 했던 부분인데 의견 감사합니다

File: cnn_test_annotated.ipynb [hunk 4]
```diff
@@ -0,0 +1,853 @@
...
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1739944463736
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# 📌 GPU 사용 가능하면 GPU로 이동 (그렇지 않으면 CPU 사용)\n",
+    "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")\n",
```
Comment (YiHeeJu) [inline cnn_test_annotated.ipynb:439]: GPU와 CPU중에 어떤 것을 사용하셔서 학습하셨나요?
GPU를 사용해서 학습된건지 CPU를 사용해서 학습된건지 정확히 출력되어있지 않아 궁금합니다:)

File: cnn_test_annotated.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (yoplnaa) [reply cnn_test_annotated.ipynb:339]: 저도 복사 붙여넣기 하는 과정에서 import가 중복되는 일이 생겼습니다.
개인적인 생각으로 진행 과정 중 따로 때어내고 작업을 진행하지는 않을 거 같아 중복을 피하는 방향으로 수정하려합니다.

File: cnn_test_annotated.ipynb [hunk 5]
```diff
@@ -0,0 +1,853 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 테스트 구동환경\n",
+    "azure notebook,  python3.8-pytorch and Tensorflow "
```
Comment (YiHeeJu) [inline cnn_test_annotated.ipynb:8]: 코드상에서는 PyTorch만 사용하고 있고 TensorFlow는 쓰이지 않은 것 같은데
혹시 테스트 구동환경에서 Tensorflow라고 명시한 이유가 따로 있으실까요?:)

File: cnn_test_annotated.ipynb (hunk 4 above)
Comment (yoplnaa) [reply cnn_test_annotated.ipynb:439]: cpu를 사용한 것입니다!

print(f"🖥️ 현재 사용 중인 디바이스: {device}") 를 추가하시면 

gpu 가능할  경우-> 🖥️ 현재 사용 중인 디바이스: cuda
cpu 사용중일 경우->🖥️ 현재 사용 중인 디바이스: cpu
와 같이 출려됩니다

File: cnn_test_annotated.ipynb [hunk 6]
```diff
@@ -0,0 +1,853 @@
...
+    "train_path = \"/mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-6a025-ml/code/Users/6a025/cnn_test/data_set/sample_data/train\"\n",
+    "test_path = \"/mnt/batch/tasks/shared/LS_root/mounts/clusters/compute-6a025-ml/code/Users/6a025/cnn_test/data_set/sample_data/test\"\n",
+    "\n",
+    "# 1️⃣ 폴더 존재 여부 확인\n",
+    "print(f\"Train 폴더 존재 여부: {os.path.exists(train_path)}\")  # 학습 데이터 폴더가 존재하는지 확인\n",
+    "print(f\"Test 폴더 존재 여부: {os.path.exists(test_path)}\")  # 테스트 데이터 폴더가 존재하는지 확인\n",
+    "\n",
+    "# 2️⃣ 이미지 변환(전처리) 설정\n",
+    "transform = transforms.Compose([\n",
+    "    transforms.Resize((128, 128)),  # 모든 이미지를 128x128 크기로 변환 (모델 입력 크기 통일)\n",
```
Comment (BrianK64) [inline cnn_test_annotated.ipynb:224]: 이미지 사이즈를  128x128픽셀로 정하면 다른 사이즈로 정했을때보다 모델이 이미지 처리할 때 정확도가 더 높아지는지 궁금합니다.

File: cnn_test_annotated.ipynb [hunk 7]
```diff
@@ -0,0 +1,853 @@
...
+    "\n",
+    "주요 기능:연산량 감소 → 모델이 더 가벼워짐\n",
+    "과적합(Overfitting) 방지 → 특정 특징이 너무 강조되는 것을 방지\n",
+    "위치 변화에 대한 불변성(Translation Invariance) 유지\n",
+    "\n",
+    "4. **완전 연결(Fully Connected, FC) 레이어**:\n",
+    "CNN의 마지막 단계로, 합성곱 레이어에서 추출된 고유한 특징을 기반으로 최종 분류를 수행하는 역할\n",
+    "일반적인 다층 퍼셉트론(MLP) 구조와 동일\n",
+    "\n",
+    "Flatten 연산을 사용하여 2D → 1D 벡터 변환 후 사용"
```
Comment (GideokKim) [inline cnn_test_annotated.ipynb:418]: 여기 정리된 마크다운이 웹에서 리뷰할때면 가독성이 떨어지는 것 같습니다. 제목이나 소제목에 줄바꿈이 적용 안되는 것 같아요!

image

File: cnn_test_annotated.ipynb (hunk 5 above)
Comment (yoplnaa) [reply cnn_test_annotated.ipynb:8]: 단순히 "python3.8-pytorch and Tensorflow"까지가 azure의 커넬 이름입니다

File: cnn_test_annotated.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (YiHeeJu) [reply cnn_test_annotated.ipynb:339]: @yoplnaa 님 말씀대로 import를 한번만 하는 것이 깔끔할 것 같습니다.
가능한 import문들을 모아서 정리해 주는 것이 가장 깔끔하다고 생각합니다:)

File: cnn_test_annotated.ipynb [hunk 8]
```diff
@@ -0,0 +1,853 @@
...
+      "Epoch [17/20], Loss: 0.0002\n",
+      "Epoch [18/20], Loss: 0.0002\n",
+      "Epoch [19/20], Loss: 0.0002\n",
+      "Epoch [20/20], Loss: 0.0001\n",
+      "✅ 학습 완료!\n"
+     ]
+    }
+   ],
+   "source": [
+    "num_epochs = 20  # 학습할 에포크 수\n",
```
Comment (BrianK64) [inline cnn_test_annotated.ipynb:534]: 혹시 에포크 수를 더 늘리거나 줄여서 학습해보셨나요?
해보셨다면 정확도가 크게 바뀌는 경루가 있었나요?

**Comment by GideokKim** [comment]
그리고 `cnn_test_annotated.ipynb`의 위치를 변경하는게 좋을 것 같습니다. 현재 루트 디렉터리에 있는데, `experiments` 디렉터리를 만드시고 그 안에 `cnn_test_annotated.ipynb`를 넣으시면 더 보기 좋을 것 같습니다!

File: cnn_test_annotated.ipynb (hunk 7 above)
Comment (yoplnaa) [reply cnn_test_annotated.ipynb:418]: 수정하겠습니다

File: cnn_test_annotated.ipynb (hunk 6 above)
Comment (yoplnaa) [reply cnn_test_annotated.ipynb:224]: 기본적으로는 데이터 셋을 일관되게 받기 위해 사이즈를 제한 합니다!
저도 방금 찾아 봤는데 연산량과 성능 최적화와 관련이 있다고 합니다.

이미지가 너무 크면 → 연산량(메모리 사용량) 증가 ⬆, 학습 속도 느림 ⏳
이미지가 너무 작으면 → 정보 손실 ⬆, 분류 정확도 낮아짐 📉
일반적으로 224x224, 128x128, 64x64 같은 크기를 많이 사용함

또한 특징 추출(Feature Extraction) 영향을 끼쳐서 어느사이즈가 성능이 좋을 직접 돌려보고 결정해야 할 거 같습니다.

너무 작은 이미지 (32×32 이하) → 중요 특징이 손실될 가능성이 높음.
너무 큰 이미지 (1024×1024 이상) → 세부 정보를 유지하지만, CNN이 특징을 효과적으로 추출하지 못할 수도 있음.

File: cnn_test_annotated.ipynb (hunk 8 above)
Comment (yoplnaa) [reply cnn_test_annotated.ipynb:534]: 에포크를 10에서 20으로 늘린 건데 4~5% 정도 정확도가 상승했습니다.
그러나 에포크가 증가해서 loss function이 거의 차이가 없어서 개인적으로는 에포크를 늘리기 보다는 데이터 증강이나 기타 하이퍼파라메터를 먼저 조정하는 쪽이 좋을 거 같습니다

**Review by BrianK64 — APPROVED** [review]

**Review by GideokKim — APPROVED** [review]

**Review by YiHeeJu — APPROVED** [review]

**Review by Yoonsoyoung02 — APPROVED** [review]

**Review by YiHeeJu — APPROVED** [review]

**Review by Yoonsoyoung02 — APPROVED** [review]

**Review by BrianK64 — APPROVED** [review]
